### PR TITLE
Add Windows to GitHub Action CI

### PR DESCRIPTION
**Pull request by Andrew-Chen-Wang**
## What do these changes do?

Remove appveyor from CI to use GitHub actions

## Are there changes in behavior for the user?

no

## Related issue number

#822

## Checklist

- [ ] I think the code is well written
- [X] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [X] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is `<Name> <Surname>`.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the [`CHANGES/`](../tree/master/CHANGES) folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example:
   `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by Andrew-Chen-Wang** [comment]
hmm not sure why the test is not running... maybe a syntax error?

**Comment by seandstewart** [comment]
Hm - I'd like to actually trigger a full run on this branch before merging.

**Comment by asvetlov** [comment]
The failure reason is: `matrix.include` doesn't provide `os` parameter.

**Comment by Andrew-Chen-Wang** [comment]
Sorry, this was meant to be a draft pull request. I just couldn't see the CI running at all without making it into a full fledged PR. You can all update the branch as I won't have time for the rest of the week.

**Comment by seandstewart** [comment]
@Andrew-Chen-Wang looks like we both sent some commits in - whoops! I've split Windows into its own job, since it's going to require some more customization and I think it will be easier to understand.

**Comment by Andrew-Chen-Wang** [comment]
Yea I was thinking it looked to clunky and wanted to rewrite it. Thanks!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -106,23 +109,81 @@ jobs:
...
+      with:
+        path: tests/requirements.txt
+    - name: Install Self (windows)
+      if: ${{ startsWith(runner.os, 'Windows') }}
+      run: |
+        pip install -e . -c tests\requirements.txt
+    - name: Run unittests
+      if: ${{ startsWith(runner.os, 'Windows') }}
+      run: |
+        python -m pytest -svv --cov --cov-report=xml --junitxml=coverage.xml
```
Comment (Andrew-Chen-Wang) [inline .github/workflows/ci.yml:180]: This needs to pass in redis-server path: https://github.com/aio-libs/aioredis/blob/b51a12a15d6f34df245280bb8ca627cddbc6d052/tests/conftest.py#L215-L221

**Comment by seandstewart** [comment]
Look at that green ✅

**Comment by Andrew-Chen-Wang** [comment]
Should we also remove timeline since we use towncrier?

**Comment by seandstewart** [comment]
@Andrew-Chen-Wang sorry for the hiatus - work has been crazy. I think we can keep timeline protection, afaict it's just making sure a towncrier entry exists. We don't need one for this change, however.

**Pull request merged**